external/fv3fit/fv3fit/reservoir/utils.py

```diff
@@ -107,18 +105,41 @@ def process_batch_data(
 
     # Concatenate features, normalize and optionally convert data
     # to latent representation
+    encoder_inputs = data
+    if trim_halo:
+        pre_trimmed_data = []
+        for arr in data:
+            tmp_divider = rank_divider.get_new_zdim_rank_divider(arr.shape[-1])
+            trimmed = tmp_divider.trim_halo_from_rank_data(arr)
+            pre_trimmed_data.append(trimmed)
+        encoder_inputs = pre_trimmed_data
+
     if autoencoder is not None:
-        data_encoded = encode_columns(data, autoencoder)
+        try:
+            data_encoded = autoencoder.encode_txyz(encoder_inputs)
+        except ValueError as e:
+            # TODO: there is a chicken/egg problem here in that no
+            # specification of transforms creates an autoencoder that
+            # expects halo during training, while pre-trained encoder
```

Comment:
Oh, that's annoying... have you been loading any existing transforms in your training? Since I switched to just using a normalization without vertical dimensionality reduction I haven't used that feature. If you haven't been loading transforms either the follow-on PR would be pretty low priority.

Reply:
Yes, I've been using the trained transformer in this PR.  I trained the hybrid SST encoder without any halo, and ran into this issue.  But I agree the follow-on is low priority given that this works